tools/myst/host/exec.c

```diff
@@ -79,9 +81,20 @@ static void* _thread_func(void* arg)
     pid_t target_tid = (pid_t)syscall(SYS_gettid);
     oe_result_t res;
     long retval = -1;
+    sigset_t set;
+
+    /* create a signal set with SIGUSR2 only */
+    sigemptyset(&set);
+    sigaddset(&set, SIGUSR2);
+
+    /* block SIGUSR2 when inside the enclave */
+    sigprocmask(SIG_BLOCK, &set, NULL);
 
     res = myst_run_thread_ecall(_enclave, &retval, cookie, event, target_tid);
 
+    /* unblock SIGUSR2 when outside the enclave */
+    sigprocmask(SIG_UNBLOCK, &set, NULL);
```

Comment:
Wouldn't current pattern block the signals during the thread lifetime?

Reply:
Yes, the goal is to block all signals except when inside ``ppoll()`` or ``epoll_pwait()``. These functions take a mask to atomically unblock signals during their operation. This is the only window in which we want to get ``SIGUSR2``.